src/Mutex/Program.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Diagnostics;
+using Temporalio.Client;
+using Temporalio.Worker;
+using TemporalioSamples.Mutex;
+using TemporalioSamples.Mutex.Impl.Internal;
+
+var client = await TemporalClient.ConnectAsync(new("localhost:7233")
+{
+    LoggerFactory = LoggerFactory.Create(builder =>
+        builder.AddSimpleConsole(options => options.TimestampFormat = "[HH:mm:ss] ").SetMinimumLevel(LogLevel.Information)),
+});
+
+async Task RunWorkerAsync()
+{
+    // Cancellation token cancelled on ctrl+c
+    using var tokenSource = new CancellationTokenSource();
+    Console.CancelKeyPress += (_, eventArgs) =>
+    {
+        tokenSource.Cancel();
+        eventArgs.Cancel = true;
+    };
+
+    // Run worker until cancelled
+    Console.WriteLine("Running worker");
+
+    var mutexActivities = new MutexActivities(client);
+
+    using var worker = new TemporalWorker(
+        client,
+        new TemporalWorkerOptions(taskQueue: "workflow-mutex-sample")
+            .AddAllActivities(typeof(Activities), null)
+            .AddAllActivities(mutexActivities)
+            .AddWorkflow<WorkflowWithMutex>()
+            .AddWorkflow<MutexWorkflow>());
```

Comment:
Feel free to have a helper or extension method for registering mutex stuffs instead of a user having to know the things needed